### PR TITLE
feat: implement grpc transcode for rest transport and complete generated tests

### PR DESCRIPTION
**Pull request by kbandes**
feat: implement grpc transcode for rest transport and complete generated tests.

### CODE REVIEW

**Review by software-dov — APPROVED** [review]
LGTM, just some formatting nits

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 1]
```diff
@@ -109,7 +119,7 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
 
     {% if service.has_lro %}
 
-    @property
+    @ property
```
[the remark this answers is not in the document]
Comment (kbandes) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:122]: Done

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 2]
```diff
@@ -168,62 +179,58 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
...
-                         - make sure dotted nested fields preserved
-                         - format url and send the request
-        #}
-        {% if 'body' in method.http_opt %}
+        http_options = [
+        {%- for rule in method.http_options %}{
+            'method': '{{ rule.method }}',
+            'uri': '{{ rule.uri }}',
+            {%- if rule.body %}
+
```
[the remark this answers is not in the document]
Comment (kbandes) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:187]: done

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 3]
```diff
@@ -168,62 +179,58 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
...
             use_integers_for_enums=False
         )
-        {% else %}
-        body = {{ method.input.ident }}.to_json(
-            request,
-            use_integers_for_enums=False
-        )
-        {% endif %}
-        {% endif %}
+        {%- endif %}{# body_spec #}
```
[the remark this answers is not in the document]
Comment (kbandes) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:213]: and current ones.

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 4]
```diff
@@ -168,62 +179,58 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
...
+        ))
 
         # Send the request
-        headers = dict(metadata)
-        headers['Content-Type'] = 'application/json'
-        response = self._session.{{ method.http_opt['verb'] }}(
-            url,
+        headers=dict(metadata)
+        headers['Content-Type']='application/json'
+        response=getattr(self._session, method)(
```
[the remark this answers is not in the document]
Comment (kbandes) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:228]: done

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2 [hunk 5]
```diff
@@ -168,62 +179,58 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
...
-        {% for field in method.query_params | sort%}
-        {% if method.input.fields[field].proto3_optional %}
-        if {{ method.input.ident }}.{{ field }} in request:
-            query_params['{{ field|camel_case }}'] = request.{{ field }}
-        {% else %}
-        query_params['{{ field|camel_case }}'] = request.{{ field }}
-        {% endif %}
-        {% endfor %}
+        # Jsonify the query params
+        query_params=json.loads({{method.input.ident}}.to_json(
```
[the remark this answers is not in the document]
Comment (kbandes) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2:219]: done

**Review by software-dov — APPROVED** [review]

**Pull request merged**